chat-android/src/main/java/com/ably/chat/Room.kt

```diff
@@ -129,9 +136,11 @@ internal class DefaultRoom(
     internal val chatApi: ChatApi,
     internal val clientId: String,
     logger: Logger,
-) : Room {
+) : Room, DiscontinuityImpl(logger) {
```

Comment:
I don't really like inheritance, especially if it's simply to share some logic, I think Kotlin way is:

```suggestion
) : Room, Discontinuity by DiscontinuityImpl(logger) {
```

Reply:
Okay, simplied the implementation for Room, 
Moved discontinuity handler to lifecyclemanager -> https://github.com/ably/ably-chat-kotlin/pull/124/commits/e60dc60e0e350ea370a9b86a25e060644d8bda59
We do the same in case of chat-js : )